src/components/merchandise/MerchandiseSpecItem.tsx

```diff
@@ -76,37 +77,47 @@ const MerchandiseSpecItem: React.VFC<{
 
         {merchandiseSpec.merchandise.isPhysical && <StyledQuantity className="px-4">x{quantity}</StyledQuantity>}
 
-        {files.length > 0 && (
-          <Button
-            loading={isDownloading}
-            onClick={async () => {
-              setIsDownloading(true)
-              let counter = 0
-              files.forEach(async file => {
-                const fileKey =
-                  file.from === 'merchandise'
-                    ? `merchandise_files/${appId}/${merchandiseSpec.merchandise.id}_${file.name}`
-                    : `merchandise_files/${appId}/${orderProductId}_${file.name}`
-                const fileLink = await getFileDownloadableLink(fileKey, authToken)
-                const fileRequest = new Request(fileLink)
-
-                try {
-                  const response = await fetch(fileRequest)
-
-                  response.url &&
-                    downloadFile(file.name, { url: response.url }).then(() => {
+        {files.length > 0 &&
+          (isDownloading ? (
+            <div className="d-flex align-items-center">
+              <Progress className="mr-2" width="80px" value={downloadProgress} size="xs" colorScheme="green" />
+              {downloadProgress}%
+            </div>
+          ) : (
+            <Button
+              loading={isDownloading}
+              onClick={async () => {
+                setIsDownloading(true)
+                let counter = 0
+                for (let file of files) {
```

Comment:
NITS: set progress to zero when button click will be better.
That can avoid repeated downloads but the initial progress shows 100%.